packages/react/tooltip/src/Tooltip.tsx

```diff
@@ -330,16 +366,109 @@ interface TooltipContentProps extends TooltipContentImplProps {
+  TooltipContentHoverableElement,
+  TooltipContentHoverableProps
+>((props: ScopedProps<TooltipContentHoverableProps>, forwardedRef) => {
+  const context = useTooltipContext(TRIGGER_NAME, props.__scopeTooltip);
+  const providerContext = useTooltipProviderContext(TOOLTIP_NAME, props.__scopeTooltip);
+  const ref = React.useRef<TooltipContentHoverableElement>(null);
+  const composedRefs = useComposedRefs(forwardedRef, ref);
+  const [pointerGraceArea, setPointerGraceArea] = React.useState<Polygon | null>(null);
+  const pointerGraceTimerRef = React.useRef(0);
+
+  const { onClose } = context;
+  const content = ref.current;
+  const trigger = context.trigger;
+
+  const handleRemoveGraceArea = React.useCallback(() => {
+    setPointerGraceArea(null);
+    removeDebugArea();
+    window.clearTimeout(pointerGraceTimerRef.current);
+    providerContext.isPointerInTransitRef.current = false;
+  }, [providerContext.isPointerInTransitRef]);
+
+  const handleCreateGraceArea = React.useCallback(
+    (event: PointerEvent, target: HTMLElement) => {
+      const pointerExitPosition = { x: event.clientX, y: event.clientY };
+      const targetPoints = getPointsFromRect(target.getBoundingClientRect());
+      const graceArea = getHull([pointerExitPosition, ...targetPoints]);
+      setPointerGraceArea(graceArea);
+      createDebugArea(graceArea);
+
+      pointerGraceTimerRef.current = window.setTimeout(() => {
+        onClose();
+        handleRemoveGraceArea();
+      }, 500);
+
+      providerContext.isPointerInTransitRef.current = true;
+    },
+    [onClose, handleRemoveGraceArea, providerContext.isPointerInTransitRef]
+  );
+
+  React.useEffect(() => {
+    return () => handleRemoveGraceArea();
+  }, [handleRemoveGraceArea]);
+
+  React.useEffect(() => {
+    if (trigger && content) {
+      const handleTriggerLeave = (event: PointerEvent) => handleCreateGraceArea(event, content);
+      const handleContentLeave = (event: PointerEvent) => handleCreateGraceArea(event, trigger);
+
+      trigger.addEventListener('pointerleave', handleTriggerLeave);
+      content.addEventListener('pointerleave', handleContentLeave);
+      return () => {
+        trigger.removeEventListener('pointerleave', handleTriggerLeave);
+        content.removeEventListener('pointerleave', handleContentLeave);
+      };
+    }
+  }, [trigger, content, context.trigger, handleCreateGraceArea, handleRemoveGraceArea]);
+
+  React.useEffect(() => {
+    if (pointerGraceArea) {
+      const handleTrackPointerGrace = (event: PointerEvent) => {
+        const target = event.target as HTMLElement;
+        const isTargetTrigger = trigger?.contains(target);
+        const isTargetContent = content?.contains(target);
+        const pointerPosition = { x: event.clientX, y: event.clientY };
+        const isPointerOutsideGraceArea = !isPointInPolygon(pointerPosition, pointerGraceArea);
+
+        if (isTargetTrigger || isTargetContent) {
+          handleRemoveGraceArea();
```

Comment:
So does this mean, if you reach the target clear the grace area?

Reply:
Yes, though you asking makes me think that they should be combined into a more descriptive name